service/service.go

```diff
@@ -208,6 +213,36 @@ func New(config Config) (*Service, error) {
 			serviceResource,
 		}
 
+		logWrapConfig := logresource.DefaultWrapConfig()
+		logWrapConfig.Logger = config.Logger
+		resources, err = logresource.Wrap(resources, logWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		retryWrapConfig := retryresource.DefaultWrapConfig()
+		retryWrapConfig.BackOffFactory = func() backoff.BackOff { return backoff.NewExponentialBackOff() }
+		retryWrapConfig.Logger = config.Logger
+		resources, err = retryresource.Wrap(resources, retryWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
```

Comment:
I think the order of wrapping with retryresouce and logresource changes the log output. Did you do that consciously?

Reply:
Mh, it should be retry -> log -> metrics right?

Reply:
I don't know. Depending on what `retry` logs. If nothing then `retry -> log -> metrics`. Otherwise current composition may be ok.

Reply:
I copied this from https://github.com/giantswarm/draughtsman-operator/blob/5319e685b8400886056b8b4ba188c5e8c3c4a9bb/service/service.go#L194-L221. The retry resource logs the underlying resource and the function and the error reason why to retry. So I think having the log resource first makes sense to have the initial "proper" logging once.

Reply:
Yeah, agreed.